packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts

```diff
@@ -160,6 +177,7 @@ export class StepScalingPolicy extends CoreConstruct {
         alarmDescription: 'Upper threshold scaling alarm',
         comparisonOperator: cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
         evaluationPeriods: props.evaluationPeriods ?? 1,
+        datapointsToAlarm: props.datapointsToAlarm,
```

Comment:
CloudFormation docs state that this is an integer with minimum value 1. We should validate this prop before we pass it in here, and throw an error if validation fails.